### PR TITLE
Plug UI

### PR DESCRIPTION
**Pull request by fmassot**
This PR goal is to embed the react app in the rust binary so that it can be served directly by a searcher or any other instances. 

Tests have been added even if there are quite dumb. At least, I ensured that the app is showing something both for unit tests and for e2e tests made with [cypress](https://www.cypress.io/).
These tests are executed in the ci workflow.

Note that the Makefile will trigger the react app build before building docker images.

TODO:
- [x] plug UI to the API
- [x] embed the UI
- [x] fix tests
- [x] update CI to build react app before building the rust project
- [x] write documentation to explain how to build

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,11 +31,17 @@ jobs:
       - run: exit 0
 
   lint:
-    name: Lint (fmt, dependency check, clippy)
+    name: Lint rust & doc (fmt, dependency check, clippy)
     runs-on: ubuntu-latest
     container: public.ecr.aws/l6o9a3f9/quickwit-builder:latest
     steps:
       - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
```
Comment (fmassot) [inline .github/workflows/ci.yml:39]: I did not want to trigger js lint and tests. Hence the use of filters.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -52,13 +58,20 @@ jobs:
         with:
           command: deny
           args: check Licenses # check that all of deps have an ok license.
-      - name: install cmake
```
Comment (fmassot) [inline .github/workflows/ci.yml:55]: cmake is not no more needed as it is already installed in the docker image

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -84,17 +97,40 @@ jobs:
...
+            js:
+              - 'quickwit-ui/**'
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: stable
       - uses: Swatinem/rust-cache@v1
       - name: Compile tests
-        run: cargo test --no-run --locked
+        run: cargo test --features=postgres --no-run --locked
```
Comment (fmassot) [inline .github/workflows/ci.yml:112]: I added the `--features` to be consistent with the next command.

File: .gitignore [hunk 4]
```diff
@@ -16,3 +16,4 @@
 .vscode-license
 .cargo
 deps
+qwdata
```
Comment (fmassot) [inline .gitignore:19]: annoying `qwdata` directory when you are testing

File: quickwit-ui/.gitignore [hunk 5]
```diff
@@ -7,9 +7,8 @@
 
 # testing
 /coverage
-
-# production
-/build
+/cypress/videos
```
Comment (fulmicoton) [inline quickwit-ui/.gitignore:10]: ?

File: quickwit-ui/.gitignore [hunk 6]
```diff
@@ -7,9 +7,8 @@
...
+/cypress/screenshots
```
Comment (fulmicoton) [inline quickwit-ui/.gitignore:11]:
```suggestion
```

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**